samza-core/src/main/java/org/apache/samza/task/TaskFactoryUtil.java

```diff
@@ -147,29 +150,31 @@ private static void validateFactory(Object factory) {
     }
   }
 
-  private static StreamTask createStreamOperatorTask(Config config, ApplicationRunner runner) throws Exception {
-    Class<?> builderClass = Class.forName(config.get(StreamApplication.APP_CLASS_CONFIG));
-    StreamApplication graphBuilder = (StreamApplication) builderClass.newInstance();
-    return new StreamOperatorTask(graphBuilder, runner);
-  }
-
-  private static boolean isStreamOperatorTask(Config config) {
+  /**
+   * Create a {@link StreamApplication} from config.
+   * Throw {@link ConfigException} if the configured class name is not a StreamApplication type, or
```

Comment:
Seems checkstyle doesn't like it since the signature doesn't have a throws there (ConfigException is runtime exception). I add throws without @ below param which is pretty close.